src/plugins/data_source/server/client/configure_client.test.ts

```diff
@@ -137,7 +137,7 @@ describe('configureClient', () => {
       configureClient(dataSourceClientParams, clientPoolSetup, config, logger)
     ).rejects.toThrowError();
 
-    expect(ClientMock).toHaveBeenCalledTimes(1);
+    expect(ClientMock).not.toHaveBeenCalled();
```

Comment:
This previously made sure it was called but also that it was called only once; is that no longer true?

Reply:
no longer true. The logic was imporved. If password determined to be contaminated, there's no need to call the `Client()` to create client